recipes/pcl/all/test_package/conanfile.py

```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake
+import os
+
+
+class PclTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    short_paths = True
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        os.chdir('bin')
+        self.run('.{}pcl_test_package'.format(os.sep))
```

Comment:
```suggestion
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "pcl_test_package")
            self.run(bin_path, run_environment=True)
```

When cross building, you can't run your test package at same processor architecture.